join-welcome/handler.py

```diff
@@ -6,71 +6,79 @@
 import hmac
 from time import perf_counter
 
-import requests
-
-def handle(req):
-    """handle a request to the function
-    Args:
-        req (str): request body
-    """
-
-    log_event(req)
+def handle(event, context):
+    # TODO implement
+    log_event(event.body)
 
     if os.getenv("log_env", "0") == "1":
         log_env()
 
     r = None
     try:
-        r = json.loads(req)
+        r  = json.loads(event.body)
     except ValueError:
         sys.stderr.write("Error parsing request, invalid JSON")
-        return "Error parsing request, invalid JSON"
-        # sys.exit(1)
-
+        return {
+            "statusCode": 400,
+            "body": "Error parsing request, invalid JSON"
+        }
+    
     if "challenge" in r:
-        return challenge(r)
+        res = challenge(r)
+        return {
+            "statusCode": 200,
+            "body": res
+        }
+    if "event" not in r:
+        return {
+            "statusCode": 400,
+            "body": "Nothing to do with webhook"
+        }
+    
+    webhook_url = read_secret("slack-incoming-webhook-url")
+    signing_secret = read_secret("slack-signing-token")
 
-    if "event" in r:
-        webhook_url = read_secret("slack-incoming-webhook-url")
-        signing_secret = read_secret("slack-signing-token")
+    target_channel = os.getenv("target_channel")
+    digest = ""
+    if "Http_X_Slack_Signature" in event.headers:
+        digest = event.headers["Http_X_Slack_Signature"]
 
-        target_channel = os.getenv("target_channel")
-        digest = os.getenv("Http_X_Slack_Signature", "")
+    # Takes format of: "Http_X_Slack_Signature v0=hash"
+    slack_request_timestamp = ""
+    if "Http_X_Slack_Request_Timestamp" in event.headers:
```

Comment:
`Http_X_Slack_Request_Timestamp` -> `X-Slack-Request-Timestamp`

Reply:
Since we read this N times, let's make a const / string for it

Reply:
Shall I do it for both `Http_X_Slack_Signature` and `X_Slack_Request_Timestamp`?

Reply:
All of them need changing to the format as they appear in the HTTP header in Flask.

`Http_X_Thing_Here` -> `X-Thing-Here`

Reply:
Updated